compiler/src/dotty/tools/backend/sjs/JSCodeGen.scala

```diff
@@ -1397,19 +1397,30 @@ class JSCodeGen()(implicit ctx: Context) {
     } else /*if (isScalaJSDefinedJSClass(currentClassSym)) {
       genJSSuperCall(tree, isStat)
     } else*/ {
-      val superCall = genApplyMethodStatically(
-          genThis()(sup.span), sym, genActualArgs(sym, args))
-
-      // Initialize the module instance just after the super constructor call.
-      if (isStaticModule(currentClassSym) && !isModuleInitialized &&
-          currentMethodSym.get.isClassConstructor) {
-        isModuleInitialized = true
-        val className = encodeClassName(currentClassSym)
-        val thisType = jstpe.ClassType(className)
-        val initModule = js.StoreModule(className, js.This()(thisType))
-        js.Block(superCall, initModule)
+      if (sym.owner.isAllOf(Scala2x | Trait) && sym.isConstructor) {
```

Comment:
Could you integrate this condition in the existing `if ... else if ... else` chain, instead of putting one more layer of if/else inside the larger `else {}` block?

Reply:
Also, if `isConstructor` boils down to testing `.is(Constructor)`, then it would be more efficient to test everything in one test:
```suggestion
      if (sym.owner.isAllOf(Scala2x | Trait | Constructor)) {
```